### PR TITLE
Branch add data type field

### PR DESCRIPTION
**Pull request by riakhaitan**

### CODE REVIEW

File: src/main/java/seedu/address/model/expense/ExpenseDate.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import java.time.LocalDate;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class ExpenseDate {
+
+
+    //public static final String VALIDATION_REGEX = "^\\d{4}\\-(0?[1-9]|1[012])\\-(0?[1-9]|[12][0-9]|3[01])";
+    public static final String VALIDATION_REGEX = "\\d+";
```
Comment (talkintomato) [inline src/main/java/seedu/address/model/expense/ExpenseDate.java:12]: i think a date format constrained regrex would be more appropriate

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 2]
```diff
@@ -9,5 +9,6 @@
     public static final Prefix PREFIX_DESCRIPTION = new Prefix("d/");
     public static final Prefix PREFIX_EXPENSE_CATEGORY = new Prefix("c/");
     public static final Prefix PREFIX_AMOUNT = new Prefix("a/");
+    public static final Prefix PREFIX_EXPENSE_DATE = new Prefix("D/");
```
Comment (talkintomato) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:12]: i think we should keep the lower case consistent and not have two d commands... maybe t/ would be a possible substitute

**Review by talkintomato — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/expense/ExpenseDate.java (hunk 1 above)
Comment (riakhaitan) [reply src/main/java/seedu/address/model/expense/ExpenseDate.java:12]: The regex I tried was not working, shall try again

File: src/main/java/seedu/address/logic/parser/CliSyntax.java (hunk 2 above)
Comment (riakhaitan) [reply src/main/java/seedu/address/logic/parser/CliSyntax.java:12]: Shall change

**Review by talkintomato — APPROVED** [review]

File: src/main/java/seedu/address/model/expense/ExpenseDate.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class ExpenseDate {
+
+
+    public static final String VALIDATION_REGEX = "^((19|2[0-9])[0-9]{2})-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])$";
+    public static final String MESSAGE_CONSTRAINTS =
+            "Date should be in a valid format yyyy-mm-dd, and it should not be blank";
+
+    public final String expenseDate;
```
Comment (jltham) [inline src/main/java/seedu/address/model/expense/ExpenseDate.java:13]: I think expenseDate could be stored as java.time.LocalDate / java.time.LocalTime! Maybe not for this milestone but useful if we wanna sort by recency.

File: src/test/java/seedu/address/storage/JsonAdaptedExpenseTest.java [hunk 4]
```diff
@@ -29,28 +30,31 @@ public void toModelType_validExpenseDetails_returnsExpense() throws Exception {
...
     }
 
     @Test
     public void toModelType_nullAmount_throwsIllegalValueException() {
-        JsonAdaptedExpense expense = new JsonAdaptedExpense(VALID_DESCRIPTION, VALID_EXPENSE_CATEGORY, null);
+        JsonAdaptedExpense expense = new JsonAdaptedExpense(VALID_DESCRIPTION, VALID_EXPENSE_CATEGORY, null,
+                VALID_EXPENSE_DATE);
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Amount.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, expense::toModelType);
     }
```
Comment (jltham) [inline src/test/java/seedu/address/storage/JsonAdaptedExpenseTest.java:60]: I think can add additional test cases for invalid date / null date! Can keep in mind for v1.3!

**Review by jltham — APPROVED** [review]
LGTM, some of the changes need to be implemented by v1.3!
